lib/Sema/CSDiagnostics.cpp

```diff
@@ -8202,20 +8202,17 @@ bool UnableToInferClosureReturnType::diagnoseAsError() {
 bool UnableToInferGenericPackElementType::diagnoseAsError() {
   auto *locator = getLocator();
   auto path = locator->getPath();
-
-  const auto applyArgToParamElt =
-      (path.end() - 2)->getAs<LocatorPathElt::ApplyArgToParam>();
-  const auto packElementElt =
-      (path.end() - 1)->getAs<LocatorPathElt::PackElement>();
-  if (!applyArgToParamElt || !packElementElt) {
-    return false;
-  }
+  const auto packElementElt = (path.end() - 1)->getAs<LocatorPathElt::PackElement>();
 
   if (isExpr<NilLiteralExpr>(getAnchor())) {
     // `nil` appears as an element of generic pack params, let's record a
     // specify contextual type for nil fix.
     emitDiagnostic(diag::unresolved_nil_literal);
   } else {
+    if (llvm::any_of(getSolution().Fixes, [&locator](const ConstraintFix *fix) {
+          return fix->getLocator() == locator;
+        }))
+      return false;
```

Comment:
Does this make sense?

Reply:
I'm not sure what are you trying to archive here, this would always return `false` because locator for diagnostic comes from the fix. What I was saying in my previous comment is that we need

```
auto packElementElt = locator->getLastElementAs<PackElement>();
assert(packElementElt);
```

That that's pretty much it.

Reply:
```swift 
func patternInstantiationTupleTest1<each T>() -> (repeat Array<each T>) {}

func patternInstantiationConcreteInvalid() {
  let _: Set<Int> = patternInstantiationTupleTest1()
  // expected-error@-1 {{cannot convert value of type '(repeat Array<_>)' to specified type 'Set<Int>'}}

  let _: (Array<Int>, Set<String>) = patternInstantiationTupleTest1() // expected-error {{'(repeat Array<Int, _>)' is not convertible to '(Array<Int>, Set<String>)', tuples have a different number of elements}}
}
```
This case produces 2 fix here:
```
```

Reply:
That's fine, let's update the test for now.

Reply:
It passes tests for now. Is there any thing I need to add?

Reply:
I understand that it's passing tests but the code is nevertheless wrong and that statement on top of this comment chain has to be removed and tests updated to add new diagnostic.

Reply:
Updated. Sorry I misunderstood that